rust/projects/parallelism/tests/thread_pool.rs

```diff
@@ -0,0 +1,64 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+use kvs::thread_pool::*;
+use kvs::Result;
+
+use crossbeam_utils::sync::WaitGroup;
+
+fn spawn_counter<P: ThreadPool>() -> Result<()> {
+    const TASK_NUM: usize = 20;
+    const ADD_COUNT: usize = 1000;
+
+    let pool = P::new(4)?;
+    let wg = WaitGroup::new();
+    let counter = Arc::new(AtomicUsize::new(0));
+
+    for _ in 0..TASK_NUM {
+        let counter = Arc::clone(&counter);
+        let wg = wg.clone();
+        pool.spawn(move || {
+            for _ in 0..ADD_COUNT {
+                counter.fetch_add(1, Ordering::Relaxed);
```

Comment:
Please use SeqCst here. I don't have confidence that the pairing of relaxed and seqcst is correct here (in general I think it is incorrect), but even if it is correct here it requires an advanced understanding of atomics to know why.

When I describe atomics in the text I will strongly suggest that they just use seqcst. Anything else is too difficult to expect the audience of this course to get correct.

Reply:
I can imagine, but don't know, that the waitgroup is a barrier to instruction reordering on both sides, such that it would be correct to use relaxed in both threads, but still, please use seqcst.